src/Compilers/Test/Core/Compilation/IRuntimeEnvironment.cs

```diff
@@ -20,33 +20,16 @@
 
 namespace Roslyn.Test.Utilities
 {
-    /// <summary>
-    /// This a factory type for creating <see cref="IRuntimeEnvironment" /> instances for the current runtime.
-    /// </summary>
-    public static class RuntimeEnvironmentFactory
-    {
-        private static IRuntimeEnvironmentFactory Instance { get; } = RuntimeUtilities.GetRuntimeEnvironmentFactory();
-
-        internal static IRuntimeEnvironment Create(ModuleData mainModule, ImmutableArray<ModuleData> modules = default) =>
-            Instance.Create(mainModule, modules);
-
-        public static (string Output, string ErrorOutput) CaptureOutput(Action action, int? maxOutputLength) =>
-            Instance.CaptureOutput(action, maxOutputLength);
-    }
-
-    public interface IRuntimeEnvironmentFactory
-    {
-        IRuntimeEnvironment Create(ModuleData mainModule, ImmutableArray<ModuleData> modules);
-        (string Output, string ErrorOutput) CaptureOutput(Action action, int? maxOutputLength);
-    }
-
     /// <summary>
     /// This is used for executing a set of modules in an isolated runtime environment for the current .NET
     /// runtime.
     /// </summary>
+    /// <remarks>
+    /// Obtain instances of this interface from <see cref="RuntimeUtilities.CreateRuntimeEnvironment(ModuleData, ImmutableArray{ModuleData})"/> 
+    /// </remarks>
     public interface IRuntimeEnvironment : IDisposable
     {
-        (int ExitCode, string Output, string ErrorOutput) Execute(string[] args, int? maxOutputLength);
+        (int ExitCode, string Output, string ErrorOutput) Execute(string[] args);
```

Comment:
Previously the `maxOutputLength` parameter was used to guard againts infinite loops, seemed useful but it's removed now?

Reply:
fwiw, I've seen that guard hit locally during development, I think.